src/core/permissions/sagas/assignPermissionsToDataSet.js

```diff
@@ -84,30 +84,33 @@ function* assignPermissionsToDataSetWorker(action :SequenceAction) :Saga<*> {
     });
 
     const updates = [];
-    keys.forEach((key :List<UUID>) => {
-
-      const ace = (new AceBuilder())
-        .setPermissions(permissionTypes)
-        .setPrincipal(principal)
-        .build();
-
-      const acl = (new AclBuilder())
-        .setAces([ace])
-        .setAclKey(key)
-        .build();
-
-      const aclData = (new AclDataBuilder())
-        .setAcl(acl)
-        .setAction(ActionTypes.ADD)
-        .build();
-
-      updates.push(aclData);
+    const permissions = Map().withMutations((mutableMap) => {
+      keys.forEach((key :List<UUID>) => {
+
+        const ace = (new AceBuilder())
+          .setPermissions(permissionTypes)
+          .setPrincipal(principal)
+          .build();
+
+        const acl = (new AclBuilder())
+          .setAces([ace])
+          .setAclKey(key)
+          .build();
+
+        const aclData = (new AclDataBuilder())
+          .setAcl(acl)
+          .setAction(ActionTypes.ADD)
+          .build();
+
+        mutableMap.set(key, ace);
+        updates.push(aclData);
+      });
     });
 
     const response :WorkerResponse = yield call(updateAclsWorker, updateAcls(updates));
     if (response.error) throw response.error;
 
-    yield put(assignPermissionsToDataSet.success(action.id));
+    yield put(assignPermissionsToDataSet.success(action.id, { permissions }));
```

Comment:
No change required, just a note. It's totally cool to do `{ permissions }`, but you don't have to. Since you're only passing along one thing, you can also just pass it directly `.success(action.id, permissions)`.